### PR TITLE
Stratis docs issue 249 html

### LINKED ISSUE #208: Update D-Bus API doc to discuss format of UUIDs (simple)

### LINKED ISSUE #249: Stop documenting particular methods in the D-Bus design doc
We don't want to do this because it is really tedious.

Instead, we'll make use of the introspection data that we already generate for the stratis-cli and the tests in stratisd. We'll annotate that introspection data with comment explaining the semantics of the method. We may or may not choose to automatically incorporate these commented chunks of XML into the D-Bus manual. We can possibly modify the ```update_introspection_data``` method in the ci repo to generate precisely the docs we need, minus the comments. Right now that script generates the introspection data wrapped in Python but that basically the last step, just spewing the introspection data out in separate files should be an easy modification. We can keep the D-Bus design doc to describe general principles of the design, so it should not go away entirely.

### PR DESCRIPTION
**Pull request by mulkieran**
Start using introspection data to document our D-Bus API. The result of these changes is that the D-Bus API doc will contain no specific information about the individual interfaces. Instead, it will have links to the XML-formatted introspection data which will be posted on the Stratis website and available in the sidebar under the heading "D-Bus Introspection Files:"

Closes #249
Closes #208

### CODE REVIEW

**Review by bgurney-rh — APPROVED** [review]

File: docs/dbus/pool.xml [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    <property name="Encrypted" type="b" access="read">
+      <annotation name="org.freedesktop.DBus.Property.EmitsChangedSignal" value="const" />
+    </property>
+    <!-- True if the pool has a cache, otherwise false. -->
+    <property name="HasCache" type="b" access="read" />
+    <!--
+      b: True if the pool is encrypted using a key in the kernel keyring.
+      bs: Encryption information.
+         b: True if the kernel keyring is the same for all the block devices
+	    in the pool, otherwise false.
```
Comment (jbaublitz) [inline docs/dbus/pool.xml:209]: This appears to be in the wrong order. `b(bs)` represents `StratisResult<Option<_>>`, not the other way around.

File: docs/dbus/pool.xml [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+    -->
+    <property name="AvailableActions" type="s" access="read" />
+    <!--
+      b: True if the pool is encrypted using Clevis.
+      b(ss): Clevis encryption configuration.
+         b: True if the Clevis configuration is consistent for all the
+	    block devices in the pool, otherwise false.
+         ss: Clevis configuration.
+	   s: Clevis pin.
+	   s: Pin-specfic Clevis configuration information.
```
Comment (jbaublitz) [inline docs/dbus/pool.xml:196]: Same comment as with `KeyDescription` property.

**Review by jbaublitz — CHANGES_REQUESTED** [review]

**Comment by mulkieran** [comment]
@jbaublitz Should be corrected now.

**Review by drckeefe — APPROVED** [review]

**Review by jbaublitz — APPROVED** [review]

**Comment by mulkieran** [comment]
squashed.

**Pull request merged**